aiida/backends/general/abstractqueries.py

```diff
@@ -136,7 +136,7 @@ def _extract_formula(args, akinds, asites):
 
         # We want only the StructureData that have attributes
         if akinds is None or asites is None:
-            return None
+            return '<<UNKNOWN>>'
```

Comment:
Just to understand: why do we need to introduce this custom label?
EDIT: Oh, I see, it is already in the code... Aren't you changing the meaning of `<<UNKNOWN>>` here, though?

Anyhow, since you now already looked into this function, if you could add a few lines of documentation to its docstring that would be great.
Have you seen whether this `args` object is documented anywhere?
We're really digging in some aiida code from the "old days" here... great to clean this up a little bit.

Reply:
Yes, I think you are right. I did this because I want to make as few changes as possible.
Now, I use `NOT FOUND` for the isolated bands, and reserve `<<UNKNOW>>` for what was originally designed for (the situation when formula variable is unproperly assigned).